clients/fides-js/src/lib/consent.tsx

```diff
@@ -0,0 +1,188 @@
+  );
+  const fetchOptions: RequestInit = {
+    mode: "cors",
+  };
+  const response = await fetch(geolocationApiUrl, fetchOptions);
+
+  if (!response.ok) {
+    debugLog(
+      options.debug,
+      "Error getting location from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+
+  try {
+    const body = await response.json();
+    debugLog(
+      options.debug,
+      "Got location response from geolocation API, returning:",
+      body
+    );
+    return body;
+  } catch (e) {
+    debugLog(
+      options.debug,
+      "Error parsing response body from geolocation API, returning {}. Response:",
+      response
+    );
+    return {};
+  }
+};
+
+/**
+ * Initialize the Fides Consent Banner or Link, with optional extraOptions to override defaults.
+ *
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initFidesConsent = async (config: FidesConfig): Promise<void> => {
+  debugLog(config.options.debug, "Initializing Fides consent...");
+
+  debugLog(
+    config.options.debug,
+    "Validating Fides consent banner options...",
+    config.options
+  );
+  if (!validateBannerOptions(config)) {
+    return Promise.reject(new Error("Invalid banner options"));
+  }
+
+  if (config.options.isDisabled) {
+    debugLog(
+      config.options.debug,
+      "Fides consent banner is disabled, skipping banner initialization!"
+    );
+    return Promise.resolve();
+  }
+
+  document.addEventListener("DOMContentLoaded", () => {
+    debugLog(config.options.debug, "DOM fully loaded and parsed");
+
+    try {
+      debugLog(
+        config.options.debug,
+        "Adding Fides consent banner CSS & HTML into the DOM..."
+      );
+      if (config.options.isGeolocationEnabled) {
+        getLocation(config.options)
+          .then(() => {
+            // todo- get applicable notices using location
+          })
+          .catch(() => {
+            // if something goes wrong with location api, we still want to render notices
+          });
+      }
+      const context = getConsentContext();
+      const consentDefaults = makeConsentDefaults({
+        config: config.consent,
+        context,
+      });
+      const onAcceptAll = () => {
+        setConsentCookieAcceptAll(consentDefaults);
+      };
+      const onRejectAll = () => {
+        setConsentCookieRejectAll(consentDefaults);
+      };
+      render(
+        <ConsentBanner
+          bannerTitle={config.experience?.banner_title}
+          bannerDescription={config.experience?.banner_description}
+          confirmationButtonLabel={config.experience?.confirmation_button_label}
+          rejectButtonLabel={config.experience?.reject_button_label}
+          privacyCenterUrl={config.options.privacyCenterUrl}
+          onAcceptAll={onAcceptAll}
+          onRejectAll={onRejectAll}
+          waitBeforeShow={100}
+        />,
+        document.body
```

Comment:
I experimented with doing something like that earlier to see if it would help with the hydration issue. it didn't help there, but I agree it's probably safer to do. I'll try to cook something up!

Reply:
so from what I can tell, doing something like that may conflict with [the hydration issue](https://www.joshwcomeau.com/react/the-perils-of-rehydration/). the init script would look something like

```typescript
// Add root div for Fides components to live in
render(<div id="fides-js-root" />, document.body);

// Append to Fides root
const fidesRoot = document.getElementById(FIDES_ROOT_ID);
if (fidesRoot) {
  render(<ConsentBanner ... />, fidesRoot);
}
```

However, that triggers the hydration warning in cookie house since the script injects `fides-js-root` which the server does not expect. so then we add our workaround and make a `<FidesRoot />` component which only renders the component after the initial mount

```typescript
const FidesRoot = () => {
    const hasMounted = useHasMounted();
    if (!hasMounted) {
      return null;
    }
  return <div id="fides-js-root"></div>;
};

// Add root div for Fides components to live in
render(<FidesRoot />, document.body);

// Append to Fides root
const fidesRoot = document.getElementById(FIDES_ROOT_ID);
if (fidesRoot) {
  // fidesRoot isn't ready yet because it hasn't been mounted yet, so the banner never loads!! 🙀 
  render(<ConsentBanner ... />, fidesRoot);
}
```

I think our choices would be
* ask the customer put an empty `fides-js-root` div somewhere in their app which we append to
* do something clever to check if `FidesRoot` is ready yet before trying to append the `ConsentBanner` to it
* ask the customer to implement the hydration workaround if they are using SSR

Reply:
another option would be to do something like

```typescript
render(<FidesRoot><ConsentBanner ... /></FidesRoot>, document.body);
```

but that's kind of similar to the original implementation

Reply:
I'll make a follow-up issue!

Reply:
https://github.com/ethyca/fides/issues/3293

Reply:
ty! Also added to our follow-up ticket list in this PR diescription